src/DIRAC/MonitoringSystem/Client/DataOperationSender.py

```diff
@@ -85,21 +85,42 @@ def sendData(self, baseDict, commitFlag=False, delayedCommit=False, startTime=Fa
 
     # Call this method in order to commit all records added but not yet committed to Accounting and Monitoring
     def concludeSending(self):
+        def accountingCommitting():
```

Comment:
you do love a complicated life, don't you ? :-) 

Use python introspection, for example:

```python
class A():

  def concludeSending(self, monitoringOptions):
    print (monitoringOptions)
    def commitAccounting():
      print("Committing Accounting")
      return 0
    def commitMonitoring():
      print("Committing Monitoring")
      return 1

    for backend in monitoringOptions:
      func = locals()[f"commit{backend}"]
      res = func()
      if not res:
        return

for opt in ( ("Monitoring", "Accounting"), ("Accounting", "Monitoring"), ("Accounting",), ("Monitoring",)):
  A().concludeSending(opt)
```

```
python /tmp/com.py 
('Monitoring', 'Accounting')
Committing Monitoring
Committing Accounting
('Accounting', 'Monitoring')
Committing Accounting
('Accounting',)
Committing Accounting
('Monitoring',)
Committing Monitoring
```

And btw, the same logic should be applied to `sendData`, since you also commit there.
And you have a bug in it:
https://github.com/DIRACGrid/DIRAC/blob/a67c921f5eb8bb3e1ebbd17946df9dba499d6f57/src/DIRAC/MonitoringSystem/Client/DataOperationSender.py#L48

Should it be ?
`            if commitFlag or not delayedCommit:`


Also, maybe we can avoid 2 different flags for the same purpose ?

Reply:
That looks much smoother for sure. I also was trying to do something like that but I wasn't sure on how to do that... Now I know :+1: